target/mips/translate.c

```diff
@@ -6656,12 +6645,7 @@ static void gen_compute_branch(DisasContext *ctx, uint32_t opc,
             SET_BTARGET_CHECKED(true); // not taken -> no need to check
             goto out;
         case OPC_J:
-#ifdef TARGET_CHERI
-            ctx->hflags |= MIPS_HFLAG_BR | MIPS_HFLAG_BDS32;
-            save_cpu_state(ctx, 0);
-#else
```

Comment:
It's OK to drop these because we might be able to directly link TBs now?

Reply:
Yes. Previously the target was not a constant so it wasn't possible. This should probably have been part of #86. I'll commit it separately.